tests/src/test/kotlin/com/epam/drill/admin/e2e/AgentSingleInstanceTest.kt

```diff
@@ -31,23 +32,19 @@ class AgentSingleInstanceTest : E2ETest() {
         createSimpleAppWithUIConnection(timeout = Duration.seconds(20)) {
             val aw = AgentWrap(agentId)
             connectAgent(aw) { _, ui, agent ->
-                ui.getAgent()?.agentStatus shouldBe AgentStatus.NOT_REGISTERED
-                ui.getBuild()?.buildStatus shouldBe BuildStatus.ONLINE
-                register(agentId) { status, _ ->
-                    status shouldBe HttpStatusCode.OK
-                }
-                ui.getAgent()?.agentStatus shouldBe AgentStatus.REGISTERING
-                ui.getBuild()?.buildStatus shouldBe BuildStatus.BUSY
-                agent.`get-set-packages-prefixes`()
-                ui.getAgent()?.agentStatus shouldBe AgentStatus.REGISTERED
-                ui.getBuild()?.apply {
-                    buildStatus shouldBe BuildStatus.ONLINE
-                    instanceIds.size shouldBe 1
+                waitUntil { ui.getAgent()?.agentStatus shouldBe AgentStatus.REGISTERED }
+                waitUntil {
+                    ui.getBuild()?.apply {
```

Comment:
I suggest replacing `shouldBe` assertions with the equality check,  since these aren't _really_ assertions.

Otherwise it looks confusing - at first glance the test seems to be written in a "wrong" way (assertions should not be hidden behind conditional control flow - e.g. safe call operator)